velox/expression/FieldReference.cpp

```diff
@@ -24,9 +24,12 @@ void FieldReference::evalSpecialForm(
     const SelectivityVector& rows,
     EvalCtx& context,
     VectorPtr& result) {
-  if (result) {
+  if (context.resultShouldBePreserved(result, rows)) {
```

Comment:
should we transition any existing implementations like this to always follow the convention of generating a local result and using copyOrMove() at the end? this will avoid adding these special case handling. what do you think?